src/main/java/edu/hm/hafner/coverage/Metric.java

```diff
@@ -111,7 +116,10 @@ public static NavigableSet<Metric> getCoverageMetrics() {
                 METHOD,
                 LINE,
                 BRANCH,
-                INSTRUCTION
+                INSTRUCTION,
+                MCDC_PAIR,
+                FUNCTION,
+                FUNCTION_CALL
```

Comment:
I haven't found time to look into your sources yet (3 other parsers are in the queue 😄 ).

What I do not yet understand: what is the meaning of each of these coverage metrics? And how do these metrics relate to the existing ones?

Reply:
For the different coverages:
- Function Coverage: This should be the same as Method Coverage (ie. did this function get called at all).  I was was unable to get the function coverage metric to translate into the existing Method Coverage metric.
- Function Call Coverage: This is a measure of the number of function calls in a function/file that have been made.  So if one has 3 function calls in a particular function, have those function calls been executed
- Modified Condition/Decision Coverage (MCDC) is a type of code coverage that includes equivalence class and boundary value coverage. **MCDC pairs are used to test for independence, which means that each condition must independently affect the decision outcome**. For example, in the truth table approach, test cases that provide MC/DC are selected by identifying pairs of rows where only one condition and the decision outcome change values between the two rows. 

**Function & Function Call** coverages are required for ISO-26262 Architectural Level testing 
**MCDC Equivalence Pair** coverage is required for for DO-178B/C Level A, the highest level of ISO 26262, and other standards.

Here's a breakdown of where VectorCAST generates coverages per the standards our customers use

<img width="506" alt="Screenshot 2024-05-14 134337" src="https://github.com/jenkinsci/coverage-model/assets/18580931/1ef7721d-9aa6-464e-9034-49cb469bdf08">

Reply:
> For the different coverages:
> 
> * Function Coverage: This should be the same as Method Coverage (ie. did this function get called at all).  I was was unable to get the function coverage metric to translate into the existing Method Coverage metric.

I see. I will have a look why this is not possible...

> * Function Call Coverage: This is a measure of the number of function calls in a function/file that have been made.  So if one has 3 function calls in a particular function, have those function calls been executed

Ok, do you have an example in your tests? I did not find such a report? All reports use `functioncall-coverage` only on the class nodes but not in the line nodes.

> * Modified Condition/Decision Coverage (MCDC) is a type of code coverage that includes equivalence class and boundary value coverage. **MCDC pairs are used to test for independence, which means that each condition must independently affect the decision outcome**. For example, in the truth table approach, test cases that provide MC/DC are selected by identifying pairs of rows where only one condition and the decision outcome change values between the two rows.
> 
> Function Call and MC/DC are provided per line.
> 

When you are reading MCDC counters, do you want to show the branch counters as well in the source code view? (I've seen that you added new line mappings to the ´FileNode´. I tried to keep the memory footprint as small as possible in that class and combined line, branch and mutation coverages into a single value. This is now somewhat pointless with all the new values)

Reply:
I will check the functioncall-coverage has good test data.

Please suggest another way to get the new way to get the coverages across the plugin instead of the expanded FileNode, and I can reimplement.

Reply:
done